src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TakeBlockOnActivationSystem.java

```diff
@@ -58,16 +58,21 @@ public void onActivate(ActivateEvent event, EntityRef entity) {
         inventoryManager = new InventoryAuthoritySystem();
 
         BlockComponent blockComponent = entity.getComponent(BlockComponent.class);
-        LASTeam teamComponent = entity.getComponent(LASTeam.class);
+        LASTeam flagTeamComponent = entity.getComponent(LASTeam.class);
+
 
         EntityRef flagTaker = event.getInstigator();
-        if (teamComponent.team.equals("red")) {
+        LASTeam playerTeamComponent = flagTaker.getComponent(LASTeam.class);
+        //if the flag being taken is a red flag and the player is on the black team, let them take the flag
+        if (flagTeamComponent.team.equals("red") && playerTeamComponent.team.equals("black")) {
```

Comment:
You should introduce constants for "red" and "black" somewhere central in the L&S project. It's easy to have a typo in there, and it's hard to change the string later on.